sentence_transformers/sparse_encoder/SparseEncoder.py

```diff
@@ -1058,39 +1058,107 @@ def intersection(
+            If batch input, returns a list of lists of tuples.
         """
-        if not embeddings.is_sparse:
+        # Ensure we have a sparse tensor for efficient processing
+        if not embeddings.is_sparse and not getattr(embeddings, "is_sparse_csr", False):
             embeddings = embeddings.to_sparse()
 
-        tokenizer = self.tokenizer
-
-        if embeddings.dim() == 2:
-            return [self.decode(embeddings[i], top_k=top_k) for i in range(embeddings.size(0))]
-        elif embeddings.dim() == 1:
-            if embeddings.is_sparse or getattr(embeddings, "is_sparse_csr", False):
-                embeddings = embeddings.coalesce() if embeddings.is_sparse else embeddings
-                values = embeddings.values()
-                indices = embeddings.indices().squeeze()
-                if values.numel() == 0:
-                    return []
-                topk = min(top_k, indices.numel())
-                top_values, top_idx = torch.topk(values, topk)
-                # Convert indices to tokens
-                top_tokens = tokenizer.convert_ids_to_tokens(indices[top_idx].tolist())
-                return list(zip(top_tokens, top_values.tolist()))
+        # For a single embedding vector
+        if embeddings.dim() == 1:
+            embeddings = embeddings.coalesce() if embeddings.is_sparse else embeddings
+            values = embeddings.values()
+            indices = embeddings.indices().squeeze()
+            if values.numel() == 0:
+                return []
+
+            # Apply top-k if specified
+            if top_k is not None:
+                top_values, top_idx = torch.topk(values, min(top_k, values.numel()))
+                indices = indices[top_idx]
+                values = top_values
+
+            # Convert token IDs to strings
+            tokens = self.tokenizer.convert_ids_to_tokens(indices.tolist())
+
+            # Return a dictionary mapping tokens to weights
+            return list(zip(tokens, values.tolist()))
+
+        # For a batch of embeddings
+        elif embeddings.dim() == 2:
+            embeddings = embeddings.coalesce() if embeddings.is_sparse else embeddings
+
+            # Extract indices and values
+            indices = embeddings.indices()
+            values = embeddings.values()
+
+            if values.numel() == 0:
+                return [{}] * embeddings.size(0)
+
+            # Sample indices (first dimension) and token indices (second dimension)
+            sample_indices, token_indices = indices[0], indices[1]
+
+            # Count tokens per sample
+            sample_counts = torch.bincount(sample_indices, minlength=embeddings.size(0)).tolist()
+
+            # Apply top-k if specified
+            if top_k is not None:
+                results = []
+                start_idx = 0
+                for i, count in enumerate(sample_counts):
+                    if count == 0:
+                        results.append({})
```

Comment:
```suggestion
                        results.append([])
```
I think we need to add an empty list here, a dict doesn't make sense